src/Plugin/Field/FieldWidget/CommerceNumberWidget.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\NumberWidget;
+
+/**
+ * Plugin implementation of the 'commerce_number' widget.
+ *
+ * @FieldWidget(
+ *   id = "commerce_number",
+ *   label = @Translation("Commerce number field"),
+ *   field_types = {
+ *     "integer",
+ *     "decimal",
+ *     "float"
+ *   }
+ * )
+ */
+class CommerceNumberWidget extends NumberWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'placeholder' => '',
+      'step' => '1',
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = parent::settingsForm($form, $form_state);
+    $field_settings = $this->getFieldSettings();
+
+    $step = '1';
+    switch ($this->fieldDefinition->getType()) {
+      case 'decimal':
+        $step = (string) pow(0.1, $field_settings['scale']);
+        break;
+
+      case 'float':
+        $step = 'any';
+        break;
+    }
+
+    $element['step'] = [
```

Comment:
We should add a comment explaining why we're not using the commerce_number field (can't use both the HTML5 input element and have locale-specific inputs with commas like commerce_number allows)